Bare return detected in an anynomous function inside a function only if the latter function’s return value is named

**Describe the bug**
With `[rule.bare-return]` on, Revive emits an error message in the case detailed below, and not in another case, also detailed below, that I think should be equivalent.

**To Reproduce**
Steps to reproduce the behavior:
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run it with the following flags & configuration file:

```shell
# flags

revive ...
```

```toml
[rule.bare-return]
```

**Expected behavior**
No output.

**Logs**
`main.go:5:3: avoid using bare returns, please add return expressions`

**Desktop (please complete the following information):**
 - OS: Linux Mint 19.2 (based on Ubuntu 18.04)
 - Version of Go: 1.13.4

**Additional context**

Here is the minimum example that triggers the Revive error:
```go
package main

func f() (err error) {
	func() {
		return
	}()

	return nil
}
```

No errors are produced if the return value is not named:
```go
package main

func f() error {
	func() {
		return
	}()

	return nil
}
```

I think there should be no difference between the two cases, since the incriminated `return` is in an anonymous function that is identical in both cases.

--- a/fixtures/bare-return.go
+++ b/fixtures/bare-return.go
@@ -31,3 +31,38 @@ func bare4(a string, b int) string {
 func bare5(a string, b int) {
 	return
 }
+
+// NR tests for issue #280
+func f280_1() (err error) {
+	func() {
+		return
+	}()
+
+	return nil
+}
+
+func f280_2() (err error) {
+	func() (r int) {
+		return // MATCH /avoid using bare returns, please add return expressions/
+	}()
+
+	return nil
+}
+
+func f280_3() (err error) {
+	func() (r int) {
+		return 1
+	}()
+
+	return // MATCH /avoid using bare returns, please add return expressions/
+}
+
+func f280_4() (err error) {
+	func() (r int) {
+		return func() (r int) {
+			return // MATCH /avoid using bare returns, please add return expressions/
+		}()
+	}()
+
+	return nil
+}

--- a/rule/bare-return.go
+++ b/rule/bare-return.go
@@ -58,6 +58,13 @@ type bareReturnFinder struct {
 }
 
 func (w bareReturnFinder) Visit(node ast.Node) ast.Visitor {
+	_, ok := node.(*ast.FuncLit)
+	if ok {
+		// skip analysing function literals
+		// they will analyzed by the lintBareReturnRule.Visit method
+		return nil
+	}
+
 	rs, ok := node.(*ast.ReturnStmt)
 	if !ok {
 		return w

